db/migration/1675107812815-AutomatedCreatedUpdatedTimestamps.ts

```diff
@@ -0,0 +1,92 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+const tablesWithTimestamps = [
+    "users",
+    "entities",
+    "sources",
+    "variables",
+    "datasets",
+    "charts",
+    "tags",
+    "chart_revisions",
+    "suggested_chart_revisions",
+]
+
+const tablesWithoutTimestamps = [
+    "namespaces",
+    "chart_dimensions",
+    "chart_slug_redirects",
+    "chart_tags",
+    "dataset_files",
+    "dataset_tags",
+    "post_tags",
+    "details",
+]
+
+export class AutomatedCreatedUpdatedTimestamps1675107812815
+    implements MigrationInterface
+{
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        // Modify the existing tables that have a createdAt and updatedAt column
+        // to use the automatic CURRENT_TIMESTAMP and ON UPDATE CURRENT_TIMESTAMP
+        // values
+        for (const table of tablesWithTimestamps) {
+            await queryRunner.query(`
+                ALTER TABLE ${table}
+                    MODIFY COLUMN createdAt DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                    MODIFY COLUMN updatedAt DATETIME NULL DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP;
+            `)
+        }
+        // settings table is similar but has different column names
+        await queryRunner.query(`
+                ALTER TABLE settings
+                    MODIFY COLUMN created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
```

Comment:
By the way do we have a consensus on camelcase vs snakecase?

Reply:
My hunch was that we tend towards camelCase and snakecase is legacy but it would be good to agree on this specifically :)